metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -527,6 +528,9 @@ func createMetadataGenSpecific(client k8sclient.Interface, commonConfig *conf.C,
 		}
 		// For example for pod named redis in namespace default, the generator uses the pod watcher for pod metadata,
 		// collects all node metadata using the node watcher's store and all namespace metadata using the namespacewatcher's store.
+		// We initialise the use_kubeadm variable based on modules KubeAdm base configuration
```

Comment:
We could set this before the call of `createMetadataGenSpecific` and `createMetadataGen`

Reply:
Thanks done in here https://github.com/elastic/beats/pull/40086/files#diff-1d7b17f1ef239e6177db5ee660f2234bfd69c82e93494d4e46ee97f8a8617cfdR595